### PR TITLE
#4188 update average_product_prices function and create backup

### PR DESCRIPTION
**Pull request by metas-lc**
#4188

### CODE REVIEW

File: de.metas.fresh/de.metas.fresh.base/src/main/sql/postgresql/ddl/functions/Average_Product_Prices.sql [hunk 1]
```diff
@@ -72,18 +74,24 @@ INNER JOIN C_BPartner bp ON ic.Bill_Bpartner_ID = bp.C_BPartner_ID
 INNER JOIN C_UOM uom ON ic.C_UOM_ID = uom.C_UOM_ID
 INNER JOIN C_UOM price_uom ON ic.Price_UOM_ID = price_uom.C_UOM_ID
 
+LEFT OUTER JOIN C_UOM_Conversion uconv ON uconv.C_UOM_ID = uom.C_UOM_ID
```
Comment (metas-ts) [inline de.metas.fresh/de.metas.fresh.base/src/main/sql/postgresql/ddl/functions/Average_Product_Prices.sql]: a `C_UOM_Conversion ` might or might not have an `M_Product_ID`
please either reflect that in the join or use the `uomconvert` DB function

File: de.metas.fresh/de.metas.fresh.base/src/main/sql/postgresql/ddl/functions/Average_Product_Prices.sql [hunk 2]
```diff
@@ -72,18 +74,24 @@ INNER JOIN C_BPartner bp ON ic.Bill_Bpartner_ID = bp.C_BPartner_ID
...
+INNER JOIN M_InOut io ON iol.M_InOut_ID = io.M_InOut_ID
 LEFT OUTER JOIN PP_Order pp ON ic.Record_ID = pp.PP_Order_ID AND ic.AD_Table_ID = get_Table_ID('PP_Order')
 
 LEFT OUTER JOIN C_Order o ON ol.C_Order_ID = o.C_Order_ID
 
 WHERE 
 	ic.isSOTrx = $3
-	AND ic.DateOrdered >= $1 -- date from
-	AND ic.DateOrdered <= $2 --  date to
+	AND io.MovementDate::date >= $1 -- date from
```
Comment (metas-ts) [inline de.metas.fresh/de.metas.fresh.base/src/main/sql/postgresql/ddl/functions/Average_Product_Prices.sql]: while you tounch the code, please replace the `$n` with parameter names

File: de.metas.fresh/de.metas.fresh.base/src/main/sql/postgresql/ddl/functions/Average_Product_Prices.sql (hunk 1 above)
Comment (metas-lc) [reply de.metas.fresh/de.metas.fresh.base/src/main/sql/postgresql/ddl/functions/Average_Product_Prices.sql]: used function and show message if the conversion is missing
(unrelated) added movement qty for more visibility on price calc

**Comment by metas-ts** [comment]
..but wait for the build

**Pull request merged**